client/src/main/java/io/pravega/client/segment/impl/SegmentMetadataClientImpl.java

```diff
@@ -234,6 +237,7 @@ private void failAllInflight(Exception e) {
     
     @Override
     public long fetchCurrentStreamLength() {
+        Preconditions.checkArgument(!closed.get(), "Already closed");
```

Comment:
`Exceptions.checkNotClosed` - it will throw `ObjectClosedException` which inherits from `IllegalStateException`. This is standard throughout the codebase